docs/introduction/style-guide.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Style guide
+---
+
+# Cadl Language Style Guide
+
+This is a guide providing a recommended set of naming convention to use when writing a Cadl spec.
+
+:::info
+The guidelines in this article are used in Cadl Core libraries. You can use them, or adapt them to your needs. The primary objectives are consistency and readability within your project, team, organization, or company source code.
+:::
+
+## Naming convention
+
+| Type             | Naming                                       | Example                                          |
+| ---------------- | -------------------------------------------- | ------------------------------------------------ |
+| scalar           | camelCase                                    | `scalar uuid extends string;`                    |
+| model            | PascalCase                                   | `model Pet {}`                                   |
+| model property   | camelCase                                    | `model Pet {furColor: string}`                   |
+| enum             | PascalCase                                   | `model Pet {furColor: string}`                   |
+| enum member      | camelCase                                    | `enum Direction {up, down}`                      |
+| namespace        | PascalCase                                   | `namespace Cadl.Rest`                            |
```

Comment:
The example here is a bit confusing because "Cadl.Rest" is not PascalCase. I assume that PascalCase applies to each component of the compound name -- so is this really "Compound PascalCase"?

Reply:
yeah I guess I should update the example to show some PascalCase namespace